packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -197,10 +199,38 @@ export class Distribution extends Resource implements IDistribution {
       const originId = scope.node.uniqueId;
       const originBindConfig = origin.bind(scope, { originId });
       this.boundOrigins.push({ origin, originId, ...originBindConfig });
+      if (originBindConfig.failoverConfig) {
+        if (addingAFailoverOrigin) {
+          throw new Error('An Origin cannot use an Origin with its own failover configuration as its fallback origin!');
+        }
+        const failoverOriginId = this.addOrigin(originBindConfig.failoverConfig.failoverOrigin, true);
+        this.addOriginGroup(originBindConfig.failoverConfig.statusCodes, originId, failoverOriginId);
+      }
       return originId;
     }
   }
 
+  private addOriginGroup(statusCodes: number[] | undefined, originId: string, failoverOriginId: string): void {
+    statusCodes = statusCodes ?? [500, 502, 503, 504];
```

Comment:
*Nit:* It feels a little weird to do this here, as opposed to in the OriginGroup, but I suppose this is more "future-proof" in terms of catching it at the latest point.

Question - Is it valid for the list of status codes to be empty (i.e., in the case where you only want to failover when the connection fails)? According to https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/high_availability_origin_failover.html, 

```
When any of the following occur:
* The primary origin returns an HTTP status code that you’ve configured for failover
* CloudFront fails to connect to the primary origin
* The response from the primary origin takes too long (times out)
Then CloudFront routes the request to the secondary origin in the origin group.
```

So I _think_ that means an empty list of status codes is valid and represents only failing over on connection issues? If so, then this should be ok; if not, we'd want to guard against that.

Reply:
I tried deploying a stack with an empty `fallbackStatusCodes`, unfortunately it does not work:

```
1 validation error detected: 
Value '[]' at 'distributionConfigWithTags.distributionConfig.originGroups.items.1.member.failoverCriteria.statusCodes.items' 
failed to satisfy constraint: 
Member must have length greater than or equal to 1
(Service: AmazonCloudFront; Status Code: 400; Error Code: MalformedXML; Request ID: ee8399fa-968c-4122-8fc9-ef60cb2cf49c)
```